File: app/src/main/java/com/guillermoMartin/calculadoraviaje/CalculatorActivity.java
package com.guillermoMartin.calculadoraviaje;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class CalculatorActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_calculator);

        EditText editTextDistance = findViewById(R.id.editTextDistance);
        EditText editTextConsumption = findViewById(R.id.editTextConsumption);
        EditText editTextPrice = findViewById(R.id.editTextPrice);
        Button buttonCalculate = findViewById(R.id.buttonCalculate);
        TextView textViewResult = findViewById(R.id.textViewResult);
        Button buttonBack = findViewById(R.id.buttonBack);

        buttonCalculate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String distanceStr = editTextDistance.getText().toString();
                String consumptionStr = editTextConsumption.getText().toString();
                String priceStr = editTextPrice.getText().toString();

                if (!distanceStr.isEmpty() && !consumptionStr.isEmpty() && !priceStr.isEmpty()) {
                    double distance = Double.parseDouble(distanceStr);
                    double consumption = Double.parseDouble(consumptionStr);
                    double price = Double.parseDouble(priceStr);

                    double totalCost = (distance / 100) * consumption * price;

                    String resultText = getString(R.string.result_prefix) + String.format("%.2f", totalCost);
                    textViewResult.setText(resultText);

                    String toastText = getString(R.string.toast_result, totalCost);
                    Toast.makeText(CalculatorActivity.this, toastText, Toast.LENGTH_LONG).show();
                } else {
                    Toast.makeText(CalculatorActivity.this, "Por favor, rellena todos los campos", Toast.LENGTH_SHORT).show();
                }
            }
        });

        buttonBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }
}

File: app/src/main/java/com/guillermoMartin/calculadoraviaje/CalculatorActivity.kt
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import com.guillermoMartin.calculadoraviaje.R

class CalculatorActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_calculator)

        val editTextDistance = findViewById<EditText>(R.id.editTextDistance)
        val editTextConsumption = findViewById<EditText>(R.id.editTextConsumption)
        val editTextPrice = findViewById<EditText>(R.id.editTextPrice)
        val buttonCalculate = findViewById<Button>(R.id.buttonCalculate)
        val textViewResult = findViewById<TextView>(R.id.textViewResult)
        val buttonBack = findViewById<Button>(R.id.buttonBack)

        buttonCalculate.setOnClickListener {
            val distanceStr = editTextDistance.text.toString()
            val consumptionStr = editTextConsumption.text.toString()
            val priceStr = editTextPrice.text.toString()

            if (distanceStr.isNotEmpty() && consumptionStr.isNotEmpty() && priceStr.isNotEmpty()) {
                val distance = distanceStr.toDouble()
                val consumption = consumptionStr.toDouble()
                val price = priceStr.toDouble()

                // Fórmula: (distancia / 100) * consumo * precio
                val totalCost = (distance / 100) * consumption * price

                val resultText = getString(R.string.result_prefix) + " %.2f".format(totalCost)
                textViewResult.text = resultText

                val toastText = getString(R.string.toast_result, totalCost)
                Toast.makeText(this, toastText, Toast.LENGTH_LONG).show()
            } else {
                Toast.makeText(this, "Por favor, rellena todos los campos", Toast.LENGTH_SHORT).show()
            }
        }

        buttonBack.setOnClickListener {
            finish() // Cierra la actividad actual y vuelve a la anterior
        }
    }
}